pkg/dashboard/resource/function.go

```diff
@@ -356,6 +466,22 @@ func (fr *functionResource) getFunctionInfoFromRequest(request *http.Request) (*
 		return nil, nuclio.WrapErrBadRequest(errors.Wrap(err, "Failed to parse JSON body"))
 	}
 
+	return fr.processFunctionInfo(&functionInfoInstance, request.Header.Get("x-nuclio-project-name"))
+}
+
+func (fr *functionResource) validateUpdateInfo(functionInfo *functionInfo, function platform.Function) error {
+
+	// in the imported state, after the function has the skip-build and skip-deploy annotations removed,
+	// if the user tries to disable the function, it will in turn build and deploy the function and then disable it.
+	// so here we don't allow users to disable an imported function.
+	if functionInfo.Spec.Disable && function.GetStatus().State == functionconfig.FunctionStateImported {
+		return errors.New("cannot disable imported function")
```

Comment:
```suggestion
		return errors.New("Failed to disable function: non-deployed functions cannot be disabled)
```

Since the comment tells a unique state in which we hint the user that disabling isn't a thing for `imported` state, perhaps we need to expose that?